Source/Core/Content/ContentSystem.cpp

```diff
@@ -198,63 +202,63 @@ ContentLibrary* ContentSystem::LibraryFromDirectory(Status& status, StringParam
 
-  // If we don't already have the content output directory, then see if we have
-  // local prebuilt content that can be copied into the output directory (this
-  // is faster than building the content ourselves, if it exists).
-  if (!DirectoryExists(outputPath))
-  {
-    String versionedContentName = FilePath::GetFileName(ContentOutputPath);
-    String prebuiltContent = FilePath::Combine(PrebuiltContentPath, library->Name);
-    if (DirectoryExists(prebuiltContent))
+    // If we don't already have the content output directory, then see if we have
+    // local prebuilt content that can be copied into the output directory (this
+    // is faster than building the content ourselves, if it exists).
+    if (!DirectoryExists(outputPath))
     {
-      PlasmaPrint("Copying prebuilt content from '%s' to '%s'\n", prebuiltContent.c_str(), outputPath.c_str());
-      ZoneScopedN("PrebuiltContent");
-      ProfileScope("PrebuiltContent");
-      CopyFolderContents(outputPath, prebuiltContent);
+        String versionedContentName = FilePath::GetFileName(ContentOutputPath);
+        String prebuiltContent = FilePath::Combine(PrebuiltContentPath, library->Name);
+        if (DirectoryExists(prebuiltContent))
+        {
+            PlasmaPrint("Copying prebuilt content from '%s' to '%s'\n", prebuiltContent.c_str(), outputPath.c_str());
+            ZoneScopedN("PrebuiltContent");
+            ProfileScope("PrebuiltContent");
+            CopyFolderContents(outputPath, prebuiltContent);
+        }
+        else
+        {
+            PlasmaPrint("Prebuilt content '%s' does not exist\n", prebuiltContent.c_str());
+        }
     }
     else
     {
-      PlasmaPrint("Prebuilt content '%s' does not exist\n", prebuiltContent.c_str());
+        PlasmaPrint("Content output for '%s' exists\n", outputPath.c_str());
     }
-  }
-  else
-  {
-    PlasmaPrint("Content output for '%s' exists\n", outputPath.c_str());
-  }
 
-  Array<ContentItem*> items;
-  items.Reserve(library->ContentItems.Size());
-  items.Append(library->ContentItems.Values());
-  HandleOf<ResourcePackage> package = PL::gContentSystem->BuildContentItems(status, items, library, false);
+    Array<ContentItem*> items;
+    items.Reserve(library->ContentItems.Size());
+    items.Append(library->ContentItems.Values());
+    PL::gContentSystem->BuildContentItems(status, items, package);
 
-  String libraryPackageFile = FilePath::CombineWithExtension(outputPath, library->Name, ".pack");
-  package->Save(libraryPackageFile);
+    String libraryPackageFile = FilePath::CombineWithExtension(outputPath, library->Name, ".pack");
+    package.Save(libraryPackageFile);
 
-  if (sendEvent)
-  {
-    // Changing this to non-delayed could be a problem
-    // If it is, we need to allocate the ResourcePackage.
-    ContentSystemEvent toSend;
-    toSend.mLibrary = library;
-    toSend.mPackage = package;
-    PL::gContentSystem->DispatchEvent(Events::PackageBuilt, &toSend);
-  }
+    if (false)
```

Comment:
This will never execute. Would it be better to comment this out?

Reply:
This is currently temporary code, i intend to come back to this with a more elegant handing as there are some cases where we want to send the event.

Reply:
added a change to better handle this case